cpp/src/arrow/compute/kernels/scalar_if_else_test.cc

```diff
@@ -140,12 +157,27 @@ void CheckWithDifferentShapes(const std::shared_ptr<Array>& cond,
                 "@" + std::to_string(right_idx) + "=" + right_in.scalar()->ToString();
           } else {
             right_in = right_bcast = right;
+            trace_right += "=Array";
           }
           SCOPED_TRACE(trace_right);
 
-          ASSERT_OK_AND_ASSIGN(auto exp, IfElse(cond_bcast, left_bcast, right_bcast));
+          Datum expected;
           ASSERT_OK_AND_ASSIGN(auto actual, IfElse(cond_in, left_in, right_in));
-          AssertDatumsEqual(exp, actual, /*verbose=*/true);
+          if (mask & COND_SCALAR && mask & LEFT_SCALAR && mask & RIGHT_SCALAR) {
```

Comment:
Add parentheses to make this more readable, or `mask == (COND_SCALAR | LEFT_SCALAR | RIGHT_SCALAR)`?